lsp-mode.el

```diff
@@ -4940,16 +4940,16 @@ Others: TRIGGER-CHARS"
                                     filename (error-message-string err)))
                    (file-error (lsp-warn "Failed to process xref entry, file-error, '%s': %s"
                                          filename (error-message-string err)))))))
-    (apply #'append
-           (if (-some->> locations (cl-first) (gethash "uri"))
```

Comment:
When using multiple language servers (intelephense + psalm), `locations` can contain mixed (`Location` and `LocationLink`) elements.